cloud.ganttproject.colloboque/src/main/kotlin/cloud/ganttproject/colloboque/ColloboqueServer.kt

```diff
@@ -23,52 +23,134 @@ import biz.ganttproject.core.io.parseXmlProject
 import biz.ganttproject.core.io.walkTasksDepthFirst
 import biz.ganttproject.core.time.CalendarFactory
 import biz.ganttproject.lib.fx.SimpleTreeCollapseView
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.GlobalScope
 import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.launch
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import net.sourceforge.ganttproject.GPLogger
 import net.sourceforge.ganttproject.GanttProjectImpl
 import net.sourceforge.ganttproject.parser.TaskLoader
-import net.sourceforge.ganttproject.storage.buildInsertTaskQuery
+import net.sourceforge.ganttproject.storage.*
 import org.jooq.DSLContext
 import org.jooq.SQLDialect
 import org.jooq.conf.RenderNameCase
 import org.jooq.impl.DSL
 import java.text.DateFormat
 import java.util.*
 import javax.sql.DataSource
-import net.sourceforge.ganttproject.storage.InitRecord
-import net.sourceforge.ganttproject.storage.InputXlog
+import java.time.LocalDateTime
 
+internal typealias ProjectRefid = String
+
+class ColloboqueServerException: Exception {
+  constructor(message: String): super(message)
+  constructor(message: String, cause: Throwable?): super(message, cause)
```

Comment:
why one may want to pass a `null` cause?

Reply:
Thanks, done :man_facepalming: